### PR TITLE
chore: removing usage of timed event from api

### PR DESCRIPTION
**Pull request by obecny**
## Which problem is this PR solving?

- Removing dependency on TimedEvent from api

## Short description of the changes

- Fixes https://github.com/open-telemetry/opentelemetry-js-api/issues/37

### CODE REVIEW

File: packages/opentelemetry-test-utils/testUtils.ts [hunk 1]
```diff
@@ -126,3 +126,15 @@ export const assertPropagation = (
   );
   assert.notStrictEqual(targetSpanContext.spanId, sourceSpanContext.spanId);
 };
+
+/**
+ * Represents a timed event.
+ * A timed event is an event with a timestamp.
+ */
+export interface TimedEvent {
```
Comment (dyladan) [inline packages/opentelemetry-test-utils/testUtils.ts:134]: should be able to use the type from the `tracing` module here

Comment (obecny) [reply packages/opentelemetry-test-utils/testUtils.ts:134]: the tracing doesn't have this interface yet

Comment (dyladan) [reply packages/opentelemetry-test-utils/testUtils.ts:134]: I think it should be in tracing since it's a tracing sdk specific thing

Comment (dyladan) [reply packages/opentelemetry-test-utils/testUtils.ts:134]: if this is just temporary you can just put a comment to remove later

Comment (obecny) [reply packages/opentelemetry-test-utils/testUtils.ts:134]: but it doesn't exist there yet, besides it is just one interface needed for test utils, we can change it later or leave it as it is. I'm fine for both, but for now we cant use it.

Comment (dyladan) [reply packages/opentelemetry-test-utils/testUtils.ts:134]: ok

**Review by Flarna — APPROVED** [review]

**Review by vmarchaud — APPROVED** [review]

**Pull request merged**